src/Domain/SizeOfElement.hpp

```diff
@@ -84,15 +83,19 @@ struct SizeOfElementCompute : db::ComputeTag, SizeOfElement<VolumeDim> {
                  ::Tags::Time, domain::Tags::FunctionsOfTime>;
   using return_type = typename base::type;
 
-  static constexpr void (*function)(
+  static constexpr void function(
```

Comment:
Perhaps an expert can chime in here. We use this signature is several other places in the repo as well.

Reply:
It is valid because references are fine, even if the thing being referenced would not be.

I am not really sure why this rule is useful.  The standard gives the not-very-enlightening note: "A literal type [such as a reference] is one for which it might be possible to create an object within a constant expression.  It is not a guarantee that it is possible to create such an object, nor is it a guarantee that any object of that type will usable in a constant expression."

The reason it is done in this situation is probably a mixture of the "mark it constexpr if that is legal" school of thought and mindless copy-pasting.